alarm_control_panel-card/alarm_control_panel-card.js

```diff
@@ -48,70 +62,56 @@ class AlarmControlPanelCard extends HTMLElement {
       }
       this._arm_action = config.auto_enter.arm_action;
     }
+    if (!config.states) config.states = ['arm_away', 'arm_home'];
+    if (!config.scale) config.scale = '15px';
+    this._config = Object.assign({}, config);
 
     const root = this.shadowRoot;
     if (root.lastChild) root.removeChild(root.lastChild);
-    if (!config.states) config.states = ['arm_home', 'arm_away'];
-    if (!config.scale) config.scale = '15px';
-
-    this._card = document.createElement('ha-card');
-    const content = document.createElement('div');
-    content.id = "content";
-    this._config = Object.assign({}, config);
-    this._card.appendChild(this._style(config.style));
-    this._card.appendChild(content);
-    root.appendChild(this._card);
-    this._config = Object.assign({}, config);
   }
 
   _updateCardContent(entity) {
     const root = this.shadowRoot;
     const card = root.lastChild;
     const config = this._config;
 
-    if (!config.title) {
-      card.header = this._stateToText(this._state);
-    } else {
+    if (config.title) {
       card.header = config.title;
+      root.getElementById("state-text").innerHTML = this._stateToText(this._state);
+      root.getElementById("state-text").className = `state ${this._state}`;
+    } else {
+      card.header = this._stateToText(this._state);
     }
 
-    root.getElementById("content").innerHTML = `
-      ${this._icon()}
-      ${config.title ? `<div class='state ${this._state}'>
-        ${this._stateToText(this._state)}</div>` : ''}
-      ${this._actionButtons()}
-      ${entity.attributes.code_format ? '<paper-input label="Alarm code" type="password"></paper-input>' : ''}
-      ${this._keypad(entity)}
-    `;
-
-    this._setupActions();
-    this._setupInput();
-    this._setupKeypad();
-  }
+    root.getElementById("state-icon").setAttribute("icon",
+      this._icons[this._state] || 'mdi:shield-outline');
+    root.getElementById("state-icon").className = this._state;
 
-  _icon() {
-    return `<ha-icon icon='${this._stateToIcon(this._state)}'
-      class='${this._state}'></ha-icon>`
+    const armVisible = (this._state === 'disarmed');
+    Array.from(root.getElementById("actions").children).forEach(function(item) {
+      if (armVisible) {
+        item.style.display = (item.id === "disarm") ? 'none' : "";
```

Comment:
Why not move the condition for the item.id in the if ? Makes it a lot more readable.

Reply:
I would hide / show buttons from css by adding/ removing 'state' class at a higher level (like on the card). Instead of manually going into the dom to search for items and show/ hide, just add a class at top level and use css to do the work.

Reply:
To the first comment... Yup that makes sense. However, the second comment means there's another change so the code is no longer relevant. 

To the second comment... No sure I get what you are saying. I can think of many ways. Are you saying something like this (psudocode):
```
<div class="arm-buttons">
   buttons...
</div>
disarm button
```
Then the code is something like:
```
if (armVisible) 
  display none for disarm
  display normal for arm buttons
else
  flip of above
}
```

Reply:
```
<div class="container">
   <div class="arm-buttons">
        <button class="arm">
        <button "class="disarm">
....


.arm { display: none;}
.disarm { display: none;}

.container.armed .arm { display: block;}
.container.disarmed .disarm { display: block;}

Example:

<div class="container disarmed">
   <div class="arm-buttons">
        <button class="arm">
        <button "class="disarm">
```